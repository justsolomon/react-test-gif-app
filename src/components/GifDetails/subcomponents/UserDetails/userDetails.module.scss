@import "../../../../assets/scss/base/variables";

.user-details {
  display: flex;
  flex-direction: column;
  width: 25%;

  @media #{$media-tablet} {
    width: 50%;
    margin-top: 1rem;
  }

  @media #{$media-mobile} {
    width: 100%;
  }

  &__group {
    display: flex;
    margin-bottom: 1rem;
  }

  &__avatar {
    height: 45px;
    width: 50px;
    margin-right: 0.5rem;
    border-radius: $border-radius--sm;
  }

  &__info {
    display: flex;
    flex-direction: column;
    padding-top: 0.2rem;
  }

  &__display-name {
    font-size: $font-size--base;
    margin-bottom: 0.3rem;
    font-weight: $font-weight--bold;
  }

  &__username {
    color: $tertiary-color;
    font-size: $font-size--sm;
    font-weight: $font-weight--medium;
  }

  &__bio {
    font-size: $font-size--base;
    line-height: 1.1rem;
    margin-bottom: 1rem;
  }

  button {
    color: $dark-grey;
    margin-bottom: 1rem;

    @media #{$media-desktop-sm} {
      font-size: $font-size--base;
    }
  }

  &__website {
    margin-bottom: 1rem;
    font-weight: $font-weight--medium;
    font-size: $font-size--base;

    &-header {
      margin-bottom: 0.5rem;
    }

    &-link {
      color: $secondary-color;
      text-decoration: none;
      display: flex;
      align-items: center;

      &__text {
        width: 80%;
        overflow: hidden;
        white-space: nowrap;
        text-overflow: ellipsis;
      }

      svg {
        height: $font-size--md;
        width: $font-size--md;
        margin-right: 0.2rem;
      }

      &:hover {
        text-decoration: underline;
      }
    }
  }
}
