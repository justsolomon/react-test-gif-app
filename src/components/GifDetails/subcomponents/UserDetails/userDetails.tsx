import { Button } from "components/global/Button";
import Skeleton from "react-loading-skeleton";
import { IUser } from "types/global";
import styles from "./userDetails.module.scss";
import { ReactComponent as ExternalLinkIcon } from "assets/vectors/link.svg";

interface IUserDetails {
  user: IUser;
  source: string;
  loading: boolean;
}

const UserDetails = ({ user, source, loading }: IUserDetails) => {
  return (
    <div className={styles["user-details"]}>
      {user && !loading ? (
        <div className={styles["user-details__group"]}>
          {!loading ? (
            <img
              src={user?.avatar_url}
              alt="User avatar"
              className={styles["user-details__avatar"]}
            />
          ) : (
            <Skeleton height={40} width={40} />
          )}
          <div className={styles["user-details__info"]}>
            <p className={styles["user-details__display-name"]}>
              {loading ? <Skeleton /> : user?.display_name}
            </p>
            <p className={styles["user-details__username"]}>
              {loading ? <Skeleton /> : `@${user?.username}`}
            </p>
          </div>
        </div>
      ) : null}
      {user?.description ? (
        <p className={styles["user-details__bio"]}>{user.description}</p>
      ) : null}
      {!loading ? (
        user ? (
          <Button
            onClick={() =>
              window.open(user.profile_url, "_blank", "noreferrer")
            }
            colorScheme="secondary"
          >
            View profile on Giphy
          </Button>
        ) : null
      ) : (
        <Skeleton height={40} />
      )}
      {user?.website_url || source ? (
        <div className={styles["user-details__website"]}>
          <p className={styles["user-details__website-header"]}>Source:</p>
          <a
            target="_blank"
            rel="noreferrer"
            href={user?.website_url || source}
            className={styles["user-details__website-link"]}
          >
            <ExternalLinkIcon />
            <p className={styles["user-details__website-link__text"]}>
              {user?.website_url || source}
            </p>
          </a>
        </div>
      ) : null}
    </div>
  );
};

export default UserDetails;
