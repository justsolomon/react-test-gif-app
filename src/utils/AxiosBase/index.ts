import axios from "axios";

// Creates a base instance for all axios based request
const instance = axios.create({
  baseURL: `https://api.giphy.com/v1/`
});

instance.interceptors.request.use((config) => {
  //add api key to query params
  config.params["api_key"] = process.env.REACT_APP_API_KEY;

  return config;
});

export default instance;
