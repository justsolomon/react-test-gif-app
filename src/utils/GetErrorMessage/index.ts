import { AxiosError } from "axios";

const getErrorMessage = (err: AxiosError): string => {
  let errorMessage;
  if (err.response) errorMessage = err.response.data.message;
  else errorMessage = err.message;

  return errorMessage;
};

export default getErrorMessage;
