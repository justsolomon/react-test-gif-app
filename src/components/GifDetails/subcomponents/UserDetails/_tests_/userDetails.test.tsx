import { render } from "@testing-library/react";
import { gif } from "utils/testUtils";
import UserDetails from "../userDetails";

describe("<UserDetails />", () => {
  it("should render the user details component properly", () => {
    const { asFragment } = render(
      <UserDetails user={gif.user} source={""} loading={false} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
