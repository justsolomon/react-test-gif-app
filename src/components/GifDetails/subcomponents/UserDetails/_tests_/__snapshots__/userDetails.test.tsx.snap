// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDetails /> should render the user details component properly 1`] = `
<DocumentFragment>
  <div
    class="user-details"
  >
    <div
      class="user-details__group"
    >
      <img
        alt="User avatar"
        class="user-details__avatar"
        src="https://media3.giphy.com/avatars/roosterteeth/WSNaHIPXtMHL.jpg"
      />
      <div
        class="user-details__info"
      >
        <p
          class="user-details__display-name"
        >
          Rooster Teeth
        </p>
        <p
          class="user-details__username"
        >
          @roosterteeth
        </p>
      </div>
    </div>
    <p
      class="user-details__bio"
    >
      Premium dumb content. Home of Achievement Hunter, Funhaus, RWBY, RvB, a ton of podcasts, new stuff all the time, and the best community ever.
    </p>
    <button
      class="button  button--solid button--secondary "
      type="button"
    >
      View profile on Giphy
    </button>
    <div
      class="user-details__website"
    >
      <p
        class="user-details__website-header"
      >
        Source:
      </p>
      <a
        class="user-details__website-link"
        href="http://www.roosterteeth.com"
        rel="noreferrer"
        target="_blank"
      >
        <svg>
          link.svg
        </svg>
        <p
          class="user-details__website-link__text"
        >
          http://www.roosterteeth.com
        </p>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
